virtualizarr/kerchunk.py

```diff
@@ -191,27 +193,46 @@ def dataset_to_kerchunk_refs(ds: xr.Dataset) -> KerchunkStoreRefs:
 
 def variable_to_kerchunk_arr_refs(var: xr.Variable) -> KerchunkArrRefs:
     """
-    Create a dictionary containing kerchunk-style array references from a single xarray.Variable (which wraps a ManifestArray).
+    Create a dictionary containing kerchunk-style array references from a single xarray.Variable (which wraps either a ManifestArray or a numpy array).
 
     Partially encodes the inner dicts to json to match kerchunk behaviour (see https://github.com/fsspec/kerchunk/issues/415).
     """
     from virtualizarr.manifests import ManifestArray
 
-    marr = var.data
+    if isinstance(var.data, ManifestArray):
+        marr = var.data
 
-    if not isinstance(marr, ManifestArray):
-        raise TypeError(
-            f"Can only serialize wrapped arrays of type ManifestArray, but got type {type(marr)}"
-        )
+        arr_refs: dict[str, Union[str, List[Union[str, int]]]] = {
+            str(chunk_key): chunk_entry.to_kerchunk()
+            for chunk_key, chunk_entry in marr.manifest.entries.items()
+        }
 
-    arr_refs: dict[str, Union[str, List[Union[str, int]]]] = {
-        str(chunk_key): chunk_entry.to_kerchunk()
-        for chunk_key, chunk_entry in marr.manifest.entries.items()
-    }
+        zarray = marr.zarray
 
-    zarray_dict = marr.zarray.to_kerchunk_json()
-    arr_refs[".zarray"] = zarray_dict
+    else:
+        try:
+            np_arr = var.to_numpy()
+        except AttributeError as e:
+            raise TypeError(
+                f"Can only serialize wrapped arrays of type ManifestArray or numpy.ndarray, but got type {type(var.data)}"
+            ) from e
+
+        # This encoding is what kerchunk does when it "inlines" data, see https://github.com/fsspec/kerchunk/blob/a0c4f3b828d37f6d07995925b324595af68c4a19/kerchunk/hdf.py#L472
+        byte_data = np_arr.tobytes()
+        # TODO do I really need to encode then decode like this?
+        inlined_data = (b"base64:" + base64.b64encode(byte_data)).decode('utf-8')
+
+        # TODO can this be generalized to save individual chunks of a dask array?
+        arr_refs = {'0': inlined_data}
```

Comment:
This would be cool because we could apply the encoding to every block in parallel (using `map_blocks` with #39), and that would allow the user to choose their chunking in the kerchunk output just by rechunking this variable using xarray before serializing it.

Reply:
Also doing this for only some of the chunks would effectively implement case (2) of #62.